wagtail/models/pages.py

```diff
@@ -1592,6 +1592,13 @@ def can_move_to(self, parent):
         if not parent_is_root and parent.locale_id != self.locale_id:
             return False
 
+        # if the page has a max_count_per_parent, check for a page of this type under the destination
+        if self.max_count_per_parent is not None:
+            return (
+                parent.get_children().type(self.specific_class).not_page(self).count()
+                < self.max_count_per_parent
+            )
+
```

Comment:
This is not quite right. The check should be (pseudo-code): "can exist under parent, and if max_count_per_parent, then children of this type < max count per parent

Reply:
Hi @zerolab good point, thank you 😃 

I've updated it to check if a page _can't_ exist under a parent (return False), and then check if `max_count_parent is not None` (then run the same check as above) 😃